File: java/exception.java
package input;
import java.io.DataInputStream;
import java.io.IOException;
class Main{
public static void main(String[] args) {
        int a,b,add,sub,multi,div;
        DataInputStream i = new DataInputStream(System.in);
        try{
            System.out.println("Enter 2 Numbers : ");
            a = Integer.parseInt(i.readLine());
            b = Integer.parseInt(i.readLine());
            add = a+b;
            sub = a-b;
            multi = a*b;
            div = a/b;
            System.out.println("Addition = "+ add + " Sub = " + sub + " Multiplication = " + multi + " Division = " + div);
        }catch(IOException test){
            System.out.println(test);
        }catch(ArithmeticException e){
            System.out.println("You are dividing by zero. EDIOT!!!\n" + e.getMessage());
        }catch(NumberFormatException e){
            System.out.println("Put a Number!!! EDIOT!!!\n" + e.getMessage());
        }catch(Exception e){
            System.out.println(e);
        }
    }
}

File: cpp/exception.cpp
#include <iostream>
int main (int argc, char *argv[])
{
    int a,b;
    try {
        cout << "Enter 2 numbers : ";
        cin >> a >> b;
        if (b == 0) throw b;
        cout << "Division = " << a/b << endl;
    }
    catch (int x) {
        cout << "Value of b : " << x << endl;
        cout << "Enter no again : ";
        cin >> x;
        cout << "Division = " << a/x << endl;
    }
    return 0;
}
